flumeDevice

```diff
@@ -344,6 +396,60 @@ def queryDevice()
         sendEvent(name: "usageLastDay", value: (resp.data[0]?.lastDay && (resp.data[0]?.lastDay?.size() > 0)) ? resp.data[0].lastDay.getAt(resp.data[0].lastDay.size() - 1)?.value?.toFloat().round(2) : "n/a")
         sendEvent(name: "usageLastWeek", value: (resp.data[0]?.lastWeek && (resp.data[0]?.lastWeek?.size() > 0)) ? resp.data[0].lastWeek.getAt(resp.data[0].lastWeek.size() - 1)?.value?.toFloat().round(2) : "n/a")
         sendEvent(name: "usageLastMonth", value: (resp.data[0]?.lastMonth && (resp.data[0]?.lastMonth?.size() > 0)) ? resp.data[0].lastMonth.getAt(resp.data[0].lastMonth.size() - 1)?.value?.toFloat().round(2) : "n/a")
+
+        lookbackFlow = (resp.data[0]?.flowLookback && (resp.data[0]?.flowLookback?.size() > 0) && resp.data[0].flowLookback.getAt(resp.data[0].flowLookback.size() - 1)?.value?.toFloat() > 0) ? true : false
+        refreshFlow = (resp.data[0]?.flowRefresh && (resp.data[0]?.flowRefresh?.size() > 0) && resp.data[0].flowRefresh.getAt(resp.data[0].flowRefresh.size() - 1)?.value?.toFloat() > 0) ? true : false
+
+        setLastPollTS()
+        currentFlowTS = getLastPollTS()
+        flowDetectedTS = getFlowDetectedTS()
+        flowStatus = device.currentValue("flowStatus")
+
+        if (refreshFlow || lookbackFlow)
+        {
+            // [cgmckeever] future us, this `flowDetectedTS == 0` is an edge case
+            //              where water is flowing and preferences are saved.
+            //              `flowDetectedTS` state variable gets purged,
+            //              making runTime very large, and usually triggers an alert
+            if (flowStatus == "stopped" || flowDetectedTS == 0 )
+            {
+                setFlowDetectedTS()
+                flowDetectedTS = getFlowDetectedTS()
+                logDebug(">>>>>>>>>>>>> startFlowWatch: ${flowDetectedTS}")
+            }
+
+            if (refreshFlow)
+            {
+                status = "running"
+            } else
+            {
+                status = "monitoring"
+            }
+
+            if (flowStatus != status)
+            {
+                sendEvent(name: "flowStatus", value: status)
+            }
+
+            runTime = Math.round((currentFlowTS - flowDetectedTS)/600)/100
+            sendEvent(name: "flowDurationDetail", value: status + " for " + runTime.toString() + " min")
+            sendEvent(name: "flowDurationMin", value: runTime)
+        } else
+        {
+            if (flowStatus == null)
+            {
+                sendEvent(name: "flowDurationMin", value: 0)
+                sendEvent(name: "flowDurationDetail", value: "flow monitoring initialzing")
+                sendEvent(name: "flowStatus", value: "stopped")
+            } else if (flowStatus != "stopped")
```

Comment:
@tomwpublic I think since there was no initial flowStatus, the `(flowStatus != "stopped")` got tripped, updating the `RAN` time to the large number based off `0` start. I added L439 to catch and init this